### PR TITLE
SecurityFeature/Adding safegasLimit to avoid revert

### PR DESCRIPTION
**Pull request by TBouder**
## What it does ✨
Some transaction has been reverted because of an incorrect computation of the gasLimit. Nobody to blame, but we don't want user's to pay for a reverted transaction.
So we decided to add a 'safeGasLimit'. All the gas may not be used, but this should prevent failed transaction.

Moreover, the `0x3D980E50508CFd41a13837A60149927a11c03731` to `0xE537B5cc158EB71037D4125BDD7538421981E6AA` (triCrypto <-> 3Crypto) paths has been disabled

## How to test ✅
The gas limit should always be at least 2_000_000 for bowswap swaps

### CODE REVIEW

File: utils/actions.js [hunk 1]
```diff
@@ -80,8 +80,30 @@ export async function	swapTokens({provider, contractAddress, from, to, amount, m
...
-		const	transaction = await contract.swap(from, to, amount, minAmountOut, instructions);
+		const	gas = await contract.estimateGas.swap(
+			from,
+			to,
+			amount,
+			minAmountOut,
+			instructions,
+		);
+
+		const	safeGasLimit = ethers.BigNumber.from(2_000_000);
```
Comment (TBouder) [inline utils/actions.js:91]: @fameal Here is the safeLimit to 2_000_000

File: utils/actions.js [hunk 2]
```diff
@@ -80,8 +80,30 @@ export async function	swapTokens({provider, contractAddress, from, to, amount, m
...
+		let	newGasLimit = gas.add(gas.mul(50).div(100));
+		if (newGasLimit.lte(safeGasLimit)) {
+			newGasLimit = safeGasLimit;
```
Comment (TBouder) [inline utils/actions.js:94]: We are taking at least 2_000_000 (if estimate + 50% is below 2_000_000)

File: utils/actions.js [hunk 3]
```diff
@@ -80,8 +80,30 @@ export async function	swapTokens({provider, contractAddress, from, to, amount, m
...
+		}
+
+		const	transaction = await contract.swap(
+			from,
+			to,
+			amount,
+			minAmountOut,
+			instructions,
+			{gasLimit: safeGasLimit}
```
Comment (TBouder) [inline utils/actions.js:103]: Force the gasLimit to be safeGasLimit

File: utils/currentPaths.js [hunk 4]
```diff
@@ -375,7 +375,7 @@ const	CURRENT_PATHS = [
...
 	['0x3D980E50508CFd41a13837A60149927a11c03731', '0x23D3D0f1c697247d5e0a9efB37d8b0ED0C464f7f', [[False, '0x80466c64868E1ab14a1Ddf27A676C3fcBE638Fe5', 1], [True, '0x7fC77b5c7614E1533320Ea6DDc2Eb61fa00A9714', 1], [True, '0xC25099792E9349C7DD09759744ea681C7de2cb66', 1]]],
 	['0x3D980E50508CFd41a13837A60149927a11c03731', '0x8cc94ccd0f3841a468184aCA3Cc478D2148E1757', [[False, '0x80466c64868E1ab14a1Ddf27A676C3fcBE638Fe5', 0], [True, '0xbEbc44782C7dB0a1A60Cb6fe97d0b483032FF1C7', 2], [True, '0x8474DdbE98F5aA3179B3B3F5942D724aFcdec9f6', 1]]],
-	['0x3D980E50508CFd41a13837A60149927a11c03731', '0xE537B5cc158EB71037D4125BDD7538421981E6AA', [[False, '0x80466c64868E1ab14a1Ddf27A676C3fcBE638Fe5', 0], [True, '0xd51a44d3fae010294c616388b506acda1bfaae46', 0]]],
+	// ['0x3D980E50508CFd41a13837A60149927a11c03731', '0xE537B5cc158EB71037D4125BDD7538421981E6AA', [[False, '0x80466c64868E1ab14a1Ddf27A676C3fcBE638Fe5', 0], [True, '0xd51a44d3fae010294c616388b506acda1bfaae46', 0]]],
```
Comment (TBouder) [inline utils/currentPaths.js:378]: Disable this path for now

File: utils/currentPaths.js [hunk 5]
```diff
@@ -448,7 +448,7 @@ const	CURRENT_PATHS = [
 	['0xE537B5cc158EB71037D4125BDD7538421981E6AA', '0x30FCf7c6cDfC46eC237783D94Fc78553E79d4E9C', [[False, '0xd51a44d3fae010294c616388b506acda1bfaae46', 0], [True, '0xbEbc44782C7dB0a1A60Cb6fe97d0b483032FF1C7', 2], [True, '0x8038C01A0390a8c547446a0b2c18fc9aEFEcc10c', 1]]],
 	['0xE537B5cc158EB71037D4125BDD7538421981E6AA', '0x625b7DF2fa8aBe21B0A976736CDa4775523aeD1E', [[False, '0xd51a44d3fae010294c616388b506acda1bfaae46', 1], [True, '0x4CA9b3063Ec5866A4B82E437059D2C43d1be596F', 1]]],
 	['0xE537B5cc158EB71037D4125BDD7538421981E6AA', '0x3D27705c64213A5DcD9D26880c1BcFa72d5b6B0E', [[False, '0xd51a44d3fae010294c616388b506acda1bfaae46', 0], [True, '0xbEbc44782C7dB0a1A60Cb6fe97d0b483032FF1C7', 2], [True, '0x3E01dD8a5E1fb3481F0F589056b428Fc308AF0Fb', 1]]],
-	['0xE537B5cc158EB71037D4125BDD7538421981E6AA', '0x3D980E50508CFd41a13837A60149927a11c03731', [[False, '0xd51a44d3fae010294c616388b506acda1bfaae46', 0], [True, '0x80466c64868E1ab14a1Ddf27A676C3fcBE638Fe5', 0]]],
+	// ['0xE537B5cc158EB71037D4125BDD7538421981E6AA', '0x3D980E50508CFd41a13837A60149927a11c03731', [[False, '0xd51a44d3fae010294c616388b506acda1bfaae46', 0], [True, '0x80466c64868E1ab14a1Ddf27A676C3fcBE638Fe5', 0]]],
```
Comment (TBouder) [inline utils/currentPaths.js:451]: Disable this path for now

File: utils/actions.js [hunk 6]
```diff
@@ -80,8 +80,30 @@ export async function	swapTokens({provider, contractAddress, from, to, amount, m
...
+		const	gas = await contract.estimateGas.swap(
+			from,
+			to,
+			amount,
+			minAmountOut,
+			instructions,
+		);
+
+		const	safeGasLimit = ethers.BigNumber.from(2_000_000);
+		let	newGasLimit = gas.add(gas.mul(50).div(100));
```
Comment (fameal) [inline utils/actions.js:92]: simpler approach `let newGasLimit = gas.mul(150).div(100);`

**Review by fameal — APPROVED** [review]
lfg

**Pull request merged**